### PR TITLE
Feature/classloader

### PR DESCRIPTION
**Pull request by omskscream**

### CODE REVIEW

**Comment by omskscream** [comment]
@etyumentcev please review, especially questions in TODOs

**Comment by etyumentcev** [comment]
I have not found unit tests on class loader at all

**Comment by omskscream** [comment]
Tests added

**Comment by omskscream** [comment]
Now ready for review

File: src/main/java/info/smart_tools/smartactors/bootstrap/Bootstrap.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import info.smart_tools.smartactors.utils.reflection.ReflectionHelper;
+
+/**
+ * Initial application loader.
+ */
+public class Bootstrap {
+    private static Config config;
+
+    public static void main(String[] args) throws ExecutionException, InterruptedException {
```
Comment (etyumentcev) [inline src/main/java/info/smart_tools/smartactors/bootstrap/Bootstrap.java]: This class shoud not have a main method

File: src/main/java/info/smart_tools/smartactors/bootstrap/BootstrapException.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package info.smart_tools.smartactors.bootstrap;
+
+public class BootstrapException extends RuntimeException {
```
Comment (etyumentcev) [inline src/main/java/info/smart_tools/smartactors/bootstrap/BootstrapException.java]: Is it a really need? maybe we would be used one of standard exceptions?

File: src/main/java/info/smart_tools/smartactors/utils/classload/ClassesNotLoadedException.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package info.smart_tools.smartactors.utils.classload;
+
+public class ClassesNotLoadedException extends RuntimeException {
```
Comment (etyumentcev) [inline src/main/java/info/smart_tools/smartactors/utils/classload/ClassesNotLoadedException.java]: Colud we use a standard exception?

File: src/main/java/info/smart_tools/smartactors/utils/classload/FileName.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package info.smart_tools.smartactors.utils.classload;
+
+/**
+ * Represents name of versioned file
+ */
+public abstract class FileName {
```
Comment (etyumentcev) [inline src/main/java/info/smart_tools/smartactors/utils/classload/FileName.java]: I'm not sure that this class must be an abstract. I think concerete class is better

File: src/test/java/info/smart_tools/smartactors/utils/classload/ClassLoadHelperTest.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+public class ClassLoadHelperTest {
+
+    @Test(expectedExceptions = {IllegalArgumentException.class})
+    public void Should_LoadByUrlThrowException_When_UrlIsNull() {
+        ClassLoadHelper.loadByUrl(null);
+    }
+
+    @Test
+    public void Should_AddClassesToClasspath_When_UrlIsValidUrlOfJarFile() throws IOException, ClassNotFoundException, CannotCompileException {
```
Comment (etyumentcev) [inline src/test/java/info/smart_tools/smartactors/utils/classload/ClassLoadHelperTest.java]: I can't believe that there is only one test on class loading

**Pull request merged**